### PR TITLE
Improved Claim All button by adding:

### PR DESCRIPTION
**Pull request by bpop81**
* Close window when there's nothing else to claim
* Time to wait between actions was switched from the fixed 2000ms to a random number between 1500ms and 2500ms

Bumped version to 5.63

### CODE REVIEW

File: src/index.user.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 // @name		  eRepublik Stuff++ Unlocked
 // @description An unlocked version of stuff++ (https://docs.google.com/spreadsheets/d/1nal62cgC7lUmrur6NRzlPVU3uxtE59WGV9-bZcPoIw8/edit#gid=0), that for some reason didn't want to run after Zordacz ban.
 // @author		Zordacz, Humberd
-// @version		5.62
+// @version		5.63
```
Comment (Humberd) [inline src/index.user.js]: New version releases are usually done in a separate commit

File: src/index.user.js [hunk 2]
```diff
@@ -4547,25 +4547,30 @@ function hookUpDailyChallengeAutoCollect() {
...
     titleRootElement.appendChild(claimAllButtonElement);
 
     function clickHandler() {
       const claimButtonElements = document.querySelectorAll(
         ".missionWrapper:not(.alreadyClaimed) .claimButton, .rewardWrapper:not(.claimed) .claimButton"
       );
+      const closeBtn = document.querySelector('#dailyMissionsPopup > a.close.closeButton[title="Close"]');
       if (!claimButtonElements.length) {
         console.log("No more challenges to claim.");
+        closeBtn.click();
```
Comment (Humberd) [inline src/index.user.js]: What if there is no close button?
We need to handle such cases.

File: src/index.user.js [hunk 3]
```diff
@@ -4547,25 +4547,30 @@ function hookUpDailyChallengeAutoCollect() {
...
-    claimAllButtonElement.textContent = "Claim All";
+    claimAllButtonElement.textContent = "Claim All & Close";
     claimAllButtonElement.addEventListener("click", clickHandler);
     titleRootElement.appendChild(claimAllButtonElement);
 
     function clickHandler() {
       const claimButtonElements = document.querySelectorAll(
         ".missionWrapper:not(.alreadyClaimed) .claimButton, .rewardWrapper:not(.claimed) .claimButton"
       );
+      const closeBtn = document.querySelector('#dailyMissionsPopup > a.close.closeButton[title="Close"]');
```
Comment (Humberd) [inline src/index.user.js]: It would be good to stick to a naming variable convention. In our case `closeButtonElement`

File: src/index.user.js [hunk 4]
```diff
@@ -4547,25 +4547,30 @@ function hookUpDailyChallengeAutoCollect() {
...
         console.log("No more challenges to claim.");
+        closeBtn.click();
         return;
       }
       console.log(`Challenges to claim: ${claimButtonElements.length}`);
       for (let claimButtonElement of claimButtonElements) {
         claimButtonElement.click();
       }
+
+      const timeToWait = Math.floor(1500 + Math.random()*(2500 - 1500 + 1)); // Random int between 1500 and 2500
```
Comment (Humberd) [inline src/index.user.js]: I think we can even go an order of magnitude lower.
What I noticed is that erep doesn't allow clicking all the buttons at once, but after clicking one you can almost immediately click another.

Comment (Humberd) [inline src/index.user.js]: It would be very benefitial to include units in a variable name. In our case `timeToWaitInMs`.

File: src/index.user.js [hunk 5]
```diff
@@ -4547,25 +4547,30 @@ function hookUpDailyChallengeAutoCollect() {
...
         ".missionWrapper:not(.alreadyClaimed) .claimButton, .rewardWrapper:not(.claimed) .claimButton"
       );
+      const closeBtn = document.querySelector('#dailyMissionsPopup > a.close.closeButton[title="Close"]');
       if (!claimButtonElements.length) {
         console.log("No more challenges to claim.");
+        closeBtn.click();
         return;
       }
       console.log(`Challenges to claim: ${claimButtonElements.length}`);
       for (let claimButtonElement of claimButtonElements) {
```
Comment (Humberd) [inline src/index.user.js]: Erep only allows clicking one of these buttons at once, so instead of looping over all elements we can click only the first one. I know it's not in your PR, but it would be very benefitial to include this change. Can you do this?

**Review by Humberd — CHANGES_REQUESTED** [review]
Awesome PR :) I suggested possible improvements to make.

File: src/index.user.js (hunk 4 above)
Comment (bpop81) [reply src/index.user.js]: You mean something like min = 1000 and max = 2000 or even lower than that?

File: src/index.user.js (hunk 1 above)
Comment (bpop81) [reply src/index.user.js]: good to know

File: src/index.user.js (hunk 2 above)
Comment (bpop81) [reply src/index.user.js]: I hesitated bedcause I don't like if's in if's, but I'll add it

File: src/index.user.js (hunk 3 above)
Comment (bpop81) [reply src/index.user.js]: sure

File: src/index.user.js (hunk 4 above)
Comment (bpop81) [reply src/index.user.js]: will do

File: src/index.user.js (hunk 2 above)
Comment (Humberd) [reply src/index.user.js]: You can use optional chaining.

File: src/index.user.js (hunk 4 above)
Comment (Humberd) [reply src/index.user.js]: I mean 150 to 200

**Comment by Humberd** [comment]
Works great! However, I think we can decrease timeout even lower. Maybe to 200-300ms?

**Comment by bpop81** [comment]
I'm affraid not to trigger some anti spam/anti throttling protection from erep, and I was hoping to simulate a fast normal user clicking on the page, that's why I went for the random interval. I think we could keep it this way for a while for the public and I can experiment with lower values on my machine for a week or two, and if everything is ok, we can decrease the values in the next update. Either way, it's already faster than before.

**Comment by Humberd** [comment]
That's true. I remember I've been manually spam clicking for a quite some time and nothing happened to me so far. Anyway, this looks good enough :) Good job.
